evm_loader/program/src/evm/database.rs

```diff
@@ -51,3 +49,244 @@ pub trait Database {
+
+        async fn nonce(&self, address: Address, chain_id: u64) -> Result<u64> {
+            Ok(self
+                .0
+                .get(&address)
+                .map(|entry| entry.nonce)
+                .unwrap_or_default())
+        }
+
+        fn increment_nonce(&mut self, address: Address, chain_id: u64) -> Result<()> {
+            unimplemented!();
+        }
+
+        async fn balance(&self, address: Address, chain_id: u64) -> Result<U256> {
+            Ok(self
+                .0
+                .get(&address)
+                .map(|entry| entry.balance)
+                .unwrap_or_default())
+        }
+
+        async fn transfer(
+            &mut self,
+            source: Address,
+            target: Address,
+            chain_id: u64,
+            value: U256,
+        ) -> Result<()> {
+            unimplemented!();
+        }
+
+        async fn code_size(&self, address: Address) -> Result<usize> {
+            unimplemented!();
+        }
+
+        async fn code(&self, address: Address) -> Result<Buffer> {
+            Ok(self
+                .0
+                .get(&address)
+                .map(|entry| Buffer::from_slice(&entry.code))
+                .unwrap_or_default())
+        }
+
+        fn set_code(&mut self, address: Address, chain_id: u64, code: Vec<u8>) -> Result<()> {
+            unimplemented!();
+        }
+
+        fn selfdestruct(&mut self, address: Address) -> Result<()> {
+            unimplemented!();
+        }
+
+        async fn storage(&self, address: Address, index: U256) -> Result<[u8; 32]> {
+            unimplemented!();
+        }
+
+        fn set_storage(&mut self, address: Address, index: U256, value: [u8; 32]) -> Result<()> {
+            unimplemented!();
+        }
+
+        async fn block_hash(&self, number: U256) -> Result<[u8; 32]> {
+            unimplemented!();
+        }
+
+        fn block_number(&self) -> Result<U256> {
+            unimplemented!();
+        }
+
+        fn block_timestamp(&self) -> Result<U256> {
+            unimplemented!();
+        }
+
+        async fn map_solana_account<F, R>(&self, address: &Pubkey, action: F) -> R
+        where
+            F: FnOnce(&AccountInfo) -> R,
+        {
+            unimplemented!();
+        }
+
+        fn snapshot(&mut self) {
+            unimplemented!();
+        }
+
+        fn revert_snapshot(&mut self) {
+            unimplemented!();
+        }
+
+        fn commit_snapshot(&mut self) {
+            unimplemented!();
+        }
+
+        async fn precompile_extension(
+            &mut self,
+            context: &Context,
+            address: &Address,
+            data: &[u8],
+            is_static: bool,
+        ) -> Option<Result<Vec<u8>>> {
+            unimplemented!();
+        }
+    }
+
+    macro_rules! code_hash_test {
+        ($fn_name:ident, $address:expr, $expected:expr) => {
+            #[maybe_async::test(feature = "is_sync", async(not(feature = "is_sync"), tokio::test))]
```

Comment:
Using `#[tokio::test]` instead of `#[maybe_async::test(feature = "is_sync", async(not(feature = "is_sync"), tokio::test))]` is enough for this PR.

`cargo test-sbf` still does not work and it might never work, it's an open problem.